server/tests/Api.IntegrationTests/controllers/LecturesControllerTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+using Microsoft.AspNetCore.Mvc.Testing;
+using StudyZen.Application.Dtos;
+using StudyZen.Domain.Entities;
+
+namespace Api.IntegrationTests.controllers;
+
+public class LecturesControllerTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly HttpClient _httpClient;
+    private readonly UpdateLectureDto _updateLectureDto;
+
+    public LecturesControllerTests(WebApplicationFactory<Program> factory)
+    {
+        _httpClient = factory.CreateClient();
+        _updateLectureDto = new UpdateLectureDto("New name", "New cont");
+    }
+
+    private async Task<CourseDto> CreateCourse()
+    {
+        var createCourseDto = new CreateCourseDto("Test name", "Test desc");
+        var courseResponse = await _httpClient.PostAsJsonAsync("Courses", createCourseDto);
+        Assert.Equal(HttpStatusCode.Created, courseResponse.StatusCode);
+        var course = await courseResponse.Content.ReadFromJsonAsync<CourseDto>();
+        Assert.NotNull(course);
+        return course;
+    }
```

Comment:
As I understand, this is not a test but rather a method to create course for testing purposes, so there should be a separate method to "seed" data without asserts.
Similar to this from repository unit tests
`
void AddTestData()

    {

        var course1 = new Course("Course1", "Description1");

        var course2 = new Course("Course2", "Description2");

        var courses = new List<Course>

        {

            course1, course2

        };

        _dbContext.Courses.AddRange(courses);

        var lecture1 = new Lecture(course1.Id, "Lecture1", "Content1");

        var lecture2 = new Lecture(course2.Id, "Lecture2", "Content2");

        var lectures = new List<Lecture>

        {

            lecture1, lecture2

        };

        _dbContext.Lectures.AddRange(lectures);

        _dbContext.SaveChanges();
    }
`